src/components/Contexts/PapersDefinitionsProvider.jsx

```diff
@@ -4,47 +4,80 @@ import useFlag from 'cozy-flags/dist/useFlag'
 import log from 'cozy-logger'
 import { useClient } from 'cozy-client'
 import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+import Alerter from 'cozy-ui/transpiled/react/Alerter'
 
 import papersJSON from 'src/constants/papersDefinitions.json'
-import { fetchJsonFileByName } from 'src/utils/fetchJsonFileByName'
+import { fetchCustomPaperDefinitions } from 'src/utils/fetchCustomPaperDefinitions'
+import { fetchContentFileToJson } from 'src/utils/fetchContentFileToJson'
 
 const PapersDefinitionsContext = createContext()
 
-const paperConfigFilenameCustom = 'papersDefinitions.json'
-
 const PapersDefinitionsProvider = ({ children }) => {
   const client = useClient()
   const { t } = useI18n()
+  const [customPapersDefinitions, setCustomPapersDefinitions] = useState({
+    isLoaded: false,
+    name: '',
+    path: ''
+  })
+
   const customPapersDefinitionsFlag = useFlag('customPapersDefinitions')
   const [papersDefinitions, setPapersDefinitions] = useState([])
 
   useEffect(() => {
     ;(async () => {
       if (customPapersDefinitionsFlag) {
-        const paperJSONCustom = await fetchJsonFileByName(
-          client,
-          paperConfigFilenameCustom
-        )
-        if (paperJSONCustom) {
+        const {
+          paperConfigFilenameCustom,
+          appFolderPath,
+          file
+        } = await fetchCustomPaperDefinitions(client, t)
```

Comment:
no need to pass `t` anymore?

Reply:
Cette fonction fait appelle à `getOrCreateAppFolderWithReference`, qui en a besoin:
https://github.com/cozy/mespapiers/blob/feat/customPapersDef/src/utils/fetchCustomPaperDefinitions.js#L16-L19